Formula/l/lld.rb

```diff
@@ -0,0 +1,63 @@
+class Lld < Formula
+  desc "LLVM Project Linker"
+  homepage "https://lld.llvm.org/"
+  url "https://github.com/llvm/llvm-project/releases/download/llvmorg-19.1.0/llvm-project-19.1.0.src.tar.xz"
+  sha256 "5042522b49945bc560ff9206f25fb87980a9b89b914193ca00d961511ff0673c"
+  # The LLVM Project is under the Apache License v2.0 with LLVM Exceptions
+  license "Apache-2.0" => { with: "LLVM-exception" }
+  head "https://github.com/llvm/llvm-project.git", branch: "main"
+
+  livecheck do
+    formula "llvm"
+  end
+
+  depends_on "cmake" => :build
+  depends_on "llvm"
+  depends_on "zstd"
+  uses_from_macos "zlib"
+
+  # These used to be part of LLVM.
+  link_overwrite "bin/lld", "bin/ld64.lld", "bin/ld.lld", "bin/lld-link"
+  link_overwrite "include/lld/*", "lib/cmake/lld/*"
+
+  def install
+    system "cmake", "-S", "lld", "-B", "build",
+                    "-DBUILD_SHARED_LIBS=ON",
+                    "-DCMAKE_INSTALL_RPATH=#{rpath}",
+                    "-DLLD_BUILT_STANDALONE=ON",
+                    "-DLLD_VENDOR=#{tap&.user}",
+                    "-DLLVM_ENABLE_LTO=ON",
+                    "-DLLVM_INCLUDE_TESTS=OFF",
+                    "-DLLVM_USE_SYMLINKS=ON",
+                    *std_cmake_args
+    system "cmake", "--build", "build"
+    system "cmake", "--install", "build"
+  end
+
+  test do
+    (testpath/"bin/lld").write <<~BASH
+      #!/bin/bash
+      exit 1
+    BASH
+    chmod "+x", "bin/lld"
+
+    (testpath/"bin").install_symlink "lld" => "ld64.lld"
+    (testpath/"bin").install_symlink "lld" => "ld.lld"
+
+    (testpath/"test.c").write <<~C
+      #include <stdio.h>
+      int main() {
+        printf("hello, world!");
+        return 0;
+      }
+    C
+
+    # Check that the `-fuse-ld=lld` flag actually picks up LLD from PATH.
+    with_env(PATH: "#{testpath}/bin:#{ENV["PATH"]}") do
+      assert_match "linker command failed", shell_output("#{ENV.cc} -v -fuse-ld=lld test.c 2>&1", 1)
```

Comment:
On Linux, the error message is
```
error: ld returned 1 exit status
```
On macOS,
```
error: linker command failed with exit code 1 (use -v to see invocation)
```
Though maybe I should just use `clang` everywhere to make this uniform.